File: Models/FixedAsset.cs
namespace TaxComputationAPI.Models
{
    public class FixedAsset
    {
        public int Id {get;set;}
        public int CompanyId {get;set;} 
        public int YearId {get;set;}
        public int AssetId {get;set;}
        public decimal OpeningCost {get;set;}

        public decimal TransferCost {get;set;}

        public decimal CostAddition {get;set;}

        public decimal CostDisposal {get;set;}


      ///   public long CostClosing {get;set;}

        public bool IsTransferCostRemoved {get;set;}
        
       
         public decimal OpeningDepreciation {get;set;}

         public decimal TransferDepreciation {get;set;}
        
        public decimal DepreciationAddition {get;set;}

        public decimal DepreciationDisposal {get;set;}

     ///   public long DepreciationClosing {get;set;}

       public bool IsTransferDepreciationRemoved {get;set;}
        

    }
}

File: taxcomputation-spa/src/apis/FixedAsset.js
import axios from "axios";

export const fixedAssetMapping = async ({
  companyId,
  yearId,
  mappedCode,
  assetId,
  triBalanceId,
  isCost,
  openingCost,
  transferCost,
  transferDepreciation,
  isTransferCostRemoved,
  isTransferDepreciationRemoved,
  costAddition,
  costDisposal,
  openingDepreciation,
  depreciationAddition,
  depreciationDisposal
}) => {
  try {
    return await axios.post("/api/FixedAsset", {
      companyId,
      yearId,
      mappedCode,
      assetId,
      triBalanceId,
      isCost,
      openingCost,
      transferCost,
      transferDepreciation,
      isTransferCostRemoved,
      isTransferDepreciationRemoved,
      costAddition,
      costDisposal,
      openingDepreciation,
      depreciationAddition,
      depreciationDisposal
    });
  } catch (error) {
    throw error;
  }
};

export const fixedAssetViewData = async ({ companyId, year }) => {
  try {
    const { data } = await axios.get(`/api/FixedAsset/${companyId}/${year}`);
    return data;
  } catch (error) {
    throw error;
  }
};

export const fixedAssetUnmapping = async ({ id }) => {
  try {
    return await axios.put(`/api/FixedAsset/fixed-asset/${id}`);
  } catch (error) {
    throw error;
  }
};

export const fixedAssetDelete = async id => {
  try {
    const { data } = await axios.delete(`/api/FixedAsset/delete-fixed-asset/${id}`);
    return data;
  } catch (error) {
    throw error;
  }
};
